src/_P144_Vindriktning.ino

```diff
@@ -0,0 +1,379 @@
+//#######################################################################################################
+//################################## Plugin 144: Dust Sensor Ikea Vindriktning ##########################
+//#######################################################################################################
+/*
+  Plugin is based upon various open sources on the internet. Plugin uses the serial lib to access a serial port
+  This plugin was written by flashmark
+
+  This plugin reads the particle concentration from the PM1006 sensor in the Ikea Vindriktning
+  The original Ikea processor is controlling the PM1006 sensor, the plugin is eavesdropping the responses
+  DevicePin1 - RX on ESP, TX on PM1006
+  DevicePin2 - TX on ESP, RX on PM1006 (optional, currently not accessed by the plugin)
+*/
+
+
+// #include section
+#include "_Plugin_Helper.h"
+#ifdef USES_P144
+# include <ESPeasySerial.h>
+// Standard plugin defines
+#define PLUGIN_144
+#define PLUGIN_ID_144     144                   // plugin id
+#define PLUGIN_NAME_144   "Dust - PM1006(K) (Vindriktning)" // "Plugin Name" is what will be dislpayed in the selection list
+#define PLUGIN_VALUENAME1_144 "PM2.5"           // variable output of the plugin. The label is in quotation marks
+#define PLUGIN_144_DEBUG  true                  // set to true for extra log info
+
+
+//   PIN/port configuration is stored in the following:
+//   CONFIG_PIN1 - Used by plugin_Helper_serial (RX pin) 
+//   CONFIG_PIN2 - Used by plugin_Helper_serial (TX pin)
+//   CONFIG_PIN3 - Not used
+//   CONFIG_PORT - Used by the plugin_Helper_serial (serialType)
+//
+//   Custom configuration is stored in the following:
+//   PCONFIG(x)
+//   x can be between 1 - 8 and can store values between -32767 - 32768 (16 bit)
+//
+//   N.B. these are aliases for a longer less readable amount of code. See _Plugin_Helper.h
+//
+//
+//   PCONFIG_LABEL(x) is a function to generate a unique label used as HTML id to be able to match 
+//                    returned values when saving a configuration.
+
+// Make accessing specific parameters more readable in the code
+// #define P144_BAUDRATE           PCONFIG_LONG(0)
+// #define P144_BAUDRATE_LABEL     PCONFIG_LABEL(0)
+
+// States for statemachine used to decode received message
+typedef enum {
+    PM1006_HEADER,
+    PM1006_LENGTH,
+    PM1006_DATA,
+    PM1006_CHECK
+} pm1006_state_t;
+
+// Global variables TODO: attach them to the task to allow multiple instances
```

Comment:
Do you plan on adding this soon-ish?
Or should we do it later after it being merged?

Reply:
What is the preferred way to do this in ESPeasy? I like to reuse standards whenever it makes sense. Please be aware this is my first ESPeasy project although I use it already for some years from the release. As I will be at least not working on ESPeasy for a week I propose to do this later. As soon as I know how to do this right and find some for refactoring I will create a new pull request.